docs/codegen/activity-generation.md

```diff
@@ -0,0 +1,43 @@
+# Generating the Activity
+
+Similar to generating the set up for each screen, Khonshu also supports generating
+an `Activity` and the set up that is required for it. To do that add the
+`@NavHostActivity` annotation to a `@Composable` function. The composable will then
+be shown in the generated `Activity`.
+
+```kotlin
+ // a scope marker used to represent the `Activity`.
+sealed interface ExampleActivityScope
+
+@NavHostActivity(
+    scope = ExampleActivityScope::class,
+    parentScope = AppScope::class, // AppScope is the default value and can be omitted
+    stateMachine = ExampleStateMachine::class,
+    activityBaseClass = ComponentActivity::class, // the super class for the generated activity
```

Comment:
```suggestion
    activityBaseClass = ComponentActivity::class, // the super class for the generated activity, ComponentActivity is the default value and can be omitted
```

Reply:
This one doesn't have a default value right now. Partially because it lives in the common source set.